compiler/ast/dag/operator.go

```diff
@@ -157,10 +157,11 @@ type (
 	// optimized pushdown is naturally carried in the serialized DAG via
 	// each Trunk.
 	Trunk struct {
-		Kind     string      `json:"kind" unpack:""`
-		Source   Source      `json:"source"`
-		Seq      *Sequential `json:"seq"`
-		Pushdown Op          `json:"pushdown"`
+		Kind      string      `json:"kind" unpack:""`
+		Source    Source      `json:"source"`
+		Seq       *Sequential `json:"seq"`
+		Pushdown  Op          `json:"pushdown"`
+		KeyPruner Expr        `json:"key_pruner"`
```

Comment:
Unless you think we'll introduce another type of pruner, I'm for shortening this name to simply Pruner, but I don't feel strongly about that!